skill-tree/skilltree-database-script.sql

```diff
@@ -0,0 +1,90 @@
+CREATE DATABASE IF NOT EXISTS skilltree;
+USE skilltree;
+
+CREATE TABLE `users` (
+  `id` binary(16) NOT NULL,
+  `created_at` datetime(6) DEFAULT NULL,
+  `updated_at` datetime(6) DEFAULT NULL,
+  `first_name` varchar(255) NOT NULL,
+  `image_url` varchar(255) NOT NULL,
+  `last_name` varchar(255) DEFAULT NULL,
+  `rds_user_id` varchar(255) DEFAULT NULL,
+  `user_role` enum('MAVEN','MEMBER','SUPERUSER','USER') NOT NULL,
+  `created_by` binary(16) DEFAULT NULL,
+  `updated_by` binary(16) DEFAULT NULL,
+  `user_type` tinyint DEFAULT NULL,
+  PRIMARY KEY (`id`),
+  KEY `FKibk1e3kaxy5sfyeekp8hbhnim` (`created_by`),
+  KEY `FKci7xr690rvyv3bnfappbyh8x0` (`updated_by`),
+  CONSTRAINT `FKci7xr690rvyv3bnfappbyh8x0` FOREIGN KEY (`updated_by`) REFERENCES `users` (`id`),
+  CONSTRAINT `FKibk1e3kaxy5sfyeekp8hbhnim` FOREIGN KEY (`created_by`) REFERENCES `users` (`id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
+
+
+CREATE TABLE `skill` (
+  `id` binary(16) NOT NULL,
+  `created_at` datetime(6) DEFAULT NULL,
+  `updated_at` datetime(6) DEFAULT NULL,
+  `is_deleted` bit(1) NOT NULL,
+  `name` varchar(255) NOT NULL,
+  `skill_type` enum('ATOMIC','DERIVED') NOT NULL,
+  `created_by` binary(16) DEFAULT NULL,
+  `updated_by` binary(16) DEFAULT NULL,
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `uc_skill_name` (`name`),
+  KEY `FKk9ihlls00bo5ljvdgnsgf7r5w` (`created_by`),
+  KEY `FKga0cp46ei9oe50sknbkty2xh7` (`updated_by`),
```

Comment:
What is `KEY` here?

Reply:
Its for the index creation, I've not written the create query, took from the mysql table, as indexes or FK relation might get missed in case we write it manually.